diff --git a/qtbase/mkspecs/features/silent.prf b/qtbase/mkspecs/features/silent.prf
index 141e6bf..2158563 100644
--- a/qtbase/mkspecs/features/silent.prf
+++ b/qtbase/mkspecs/features/silent.prf
@@ -1,6 +1,6 @@
 !macx-xcode {
-    QMAKE_CC = @echo compiling $< && $$QMAKE_CC
-    QMAKE_CXX = @echo compiling $< && $$QMAKE_CXX
+    QMAKE_CC = $$QMAKE_CC
+    QMAKE_CXX = $$QMAKE_CXX
     !contains(QMAKE_LINK, "@:"):QMAKE_LINK = @echo linking $@ && $$QMAKE_LINK
     QMAKE_LINK_SHLIB = @echo linking $@ && $$QMAKE_LINK_SHLIB
 }
