--- a/cmake/EigenConfigureTesting.cmake	2015-10-01 09:06:10.000000000 +0200
+++ b/cmake/EigenConfigureTesting.cmake	2015-10-13 12:13:42.000000000 +0200
@@ -41,7 +41,7 @@
 
 # copy ctest properties, which currently
 # o raise the warning levels
-configure_file(${CMAKE_CURRENT_BINARY_DIR}/DartConfiguration.tcl ${CMAKE_BINARY_DIR}/DartConfiguration.tcl)
+#configure_file(${CMAKE_CURRENT_BINARY_DIR}/DartConfiguration.tcl ${CMAKE_BINARY_DIR}/DartConfiguration.tcl)
 
 # restore default CMAKE_MAKE_PROGRAM
 set(CMAKE_MAKE_PROGRAM ${CMAKE_MAKE_PROGRAM_SAVE})
