diff --git a/source/io/ufile.c b/source/io/ufile.c
index 820071f..9304ac0 100644
--- a/source/io/ufile.c
+++ b/source/io/ufile.c
@@ -59,7 +59,7 @@ finit_owner(FILE         *f,
     uprv_memset(result, 0, sizeof(UFILE));
     result->fFileno = fileno(f);
 
-#if U_PLATFORM_USES_ONLY_WIN32_API
+#if U_PLATFORM_USES_ONLY_WIN32_API && _MSC_VER < 1900
     if (0 <= result->fFileno && result->fFileno <= 2) {
         /* stdin, stdout and stderr need to be special cased for Windows 98 */
 #if _MSC_VER >= 1400
