From 706dbe9650746e91b4b8a95a24ff6f61c7cb628f Mon Sep 17 00:00:00 2001
From: Julien Waechter <julien.waechter@gmail.com>
Date: Fri, 27 Oct 2017 12:01:11 +0200
Subject: [PATCH] Add missing export

---
 Source/MediaStorageAndFileFormat/gdcmSegmentHelper.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Source/MediaStorageAndFileFormat/gdcmSegmentHelper.h b/Source/MediaStorageAndFileFormat/gdcmSegmentHelper.h
index 01ee18e83..11f1f7814 100644
--- a/Source/MediaStorageAndFileFormat/gdcmSegmentHelper.h
+++ b/Source/MediaStorageAndFileFormat/gdcmSegmentHelper.h
@@ -14,6 +14,8 @@
 #ifndef GDCMSEGMENTHELPER_H
 #define GDCMSEGMENTHELPER_H
 
+#include "gdcmTypes.h"
+
 #include <string>
 
 namespace gdcm
@@ -27,7 +29,7 @@ namespace SegmentHelper
   *
   * \see  PS 3.3 section 8.8.
   */
-struct BasicCodedEntry
+struct GDCM_EXPORT BasicCodedEntry
 {
   /**
     * \brief Constructor.
-- 
2.14.1.windows.1

