--- a/Utilities/VTK/vtkImageMapToColors16.cxx	2017-02-17 14:32:21.000000000 +0100
+++ b/Utilities/VTK/vtkImageMapToColors16.cxx	2017-03-16 13:46:45.223443700 +0100
@@ -66,9 +66,9 @@
 }
 
 //----------------------------------------------------------------------------
-unsigned long vtkImageMapToColors16::GetMTime()
+vtkMTimeType vtkImageMapToColors16::GetMTime()
 {
-  unsigned long t1, t2;
+  vtkMTimeType t1, t2;
 
   t1 = this->Superclass::GetMTime();
   if (this->LookupTable)
--- a/Utilities/VTK/vtkImageMapToColors16.h	2017-02-17 14:32:21.000000000 +0100
+++ b/Utilities/VTK/vtkImageMapToColors16.h	2017-03-16 13:46:45.231445300 +0100
@@ -85,7 +85,7 @@
 
   // Description:
   // We need to check the modified time of the lookup table too.
-  virtual unsigned long GetMTime();
+  virtual vtkMTimeType GetMTime();
 
 protected:
   vtkImageMapToColors16();
