diff --git a/libtiff/tiffconf.vc.h b/libtiff/tiffconf.vc.h
index 680f839..4efd2e0 100644
--- a/libtiff/tiffconf.vc.h
+++ b/libtiff/tiffconf.vc.h
@@ -46,14 +46,21 @@
 /* Signed 64-bit type formatter */
 #define TIFF_INT64_FORMAT "%I64d"
 
-/* Signed 64-bit type */
-#define TIFF_INT64_T signed __int64
-
+#ifdef __WIN32__
+    /* Signed 64-bit type */
+    #define TIFF_INT64_T signed __int64
+#else
+    #define TIFF_INT64_T long int
+#endif
 /* Unsigned 64-bit type formatter */
 #define TIFF_UINT64_FORMAT "%I64u"
 
-/* Unsigned 64-bit type */
-#define TIFF_UINT64_T unsigned __int64
+#ifdef __WIN32__
+    /* Unsigned 64-bit type */
+    #define TIFF_UINT64_T unsigned __int64
+#else
+    #define TIFF_UINT64_T unsigned long int
+#endif
 
 /* Signed size type */
 #if defined(_WIN64)
