diff -uNr a/core/vnl/io/CMakeLists.txt b/core/vnl/io/CMakeLists.txt
--- a/core/vnl/io/CMakeLists.txt
+++ b/core/vnl/io/CMakeLists.txt
@@ -22,7 +22,7 @@
 IF(CMAKE_COMPILER_IS_GNUCXX)
   SET_SOURCE_FILES_PROPERTIES(Templates/vsl_vector_io+vcl_vector+vcl_vector+vnl_vector+double----.cxx
                               PROPERTIES
-                              COMPILE_FLAGS -ftemplate-depth-35)
+                              COMPILE_FLAGS -ftemplate-depth-40)
 ENDIF(CMAKE_COMPILER_IS_GNUCXX)
 
 ADD_LIBRARY(vnl_io ${vnl_io_sources} )
