--- a/core/vcsl/CMakeLists.txt	2014-11-14 21:40:27.000000000 +0100
+++ b/core/vcsl/CMakeLists.txt	2015-09-18 17:03:22.000000000 +0200
@@ -63,7 +63,7 @@
   AUX_SOURCE_DIRECTORY( Templates vcsl_sources )
 
   ADD_LIBRARY( vcsl ${vcsl_sources} )
-  TARGET_LINK_LIBRARIES( vcsl vnl vbl )
+  TARGET_LINK_LIBRARIES( vcsl vnl )
 
   INSTALL_TARGETS(/lib vcsl)
   INSTALL_NOBASE_HEADER_FILES(/include/vxl/core/vcsl ${vcsl_sources})
