diff -uNr a/SofaKernel/framework/sofa/core/objectmodel/Data.cpp b/SofaKernel/framework/sofa/core/objectmodel/Data.cpp
--- a/SofaKernel/framework/sofa/core/objectmodel/Data.cpp	2017-10-04 08:17:16.000000000 +0200
+++ b/SofaKernel/framework/sofa/core/objectmodel/Data.cpp	2017-10-04 08:19:20.000000000 +0200
@@ -34,7 +34,6 @@

 /// Specialization for reading strings
 template<>
-inline
 bool SOFA_CORE_API TData<std::string>::read( const std::string& str )
 {
     virtualSetValue(str);
@@ -43,7 +42,6 @@

 /// Specialization for reading booleans
 template<>
-inline
 bool SOFA_CORE_API TData<bool>::read( const std::string& str )
 {
     if (str.empty())