diff --git a/applications/sofa/gui/SofaGuiMain/CMakeLists.txt b/applications/sofa/gui/SofaGuiMain/CMakeLists.txt
index 46d28a4..1eb554d 100644
--- a/applications/sofa/gui/SofaGuiMain/CMakeLists.txt
+++ b/applications/sofa/gui/SofaGuiMain/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.1)
 project(SofaGuiMain)
 
+set(HEADER_FILES ../Main.h)
+
 add_library(${PROJECT_NAME} SHARED ../Main.h ../Main.cpp)
 set(SOFAGUIMAIN_DEPS ${SOFAGUI_TARGETS})
 list(REMOVE_ITEM SOFAGUIMAIN_DEPS SofaGuiMain)
