diff --git a/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/PrimitivePositioning.cpp b/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/PrimitivePositioning.cpp
index a8a900a..5c86dee 100644
--- a/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/PrimitivePositioning.cpp
+++ b/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/PrimitivePositioning.cpp
@@ -85,6 +85,7 @@
 
 
 
+#include <algorithm>
 #include "Primitive.h"
 #include "AxisAlignedBox.h"
 #include "PrimitivePositioning.h"
diff --git a/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/Vector2.cpp b/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/Vector2.cpp
index 14aba5a..f2d7ac7 100644
--- a/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/Vector2.cpp
+++ b/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/Vector2.cpp
@@ -44,6 +44,7 @@
 
 #include "Vector2.h"
 #include "Vector3.h"
+#include <algorithm>
 #include <math.h>
 
 #ifdef WIN32
diff --git a/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/Vector3.cpp b/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/Vector3.cpp
index c629370..8a1ae40 100644
--- a/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/Vector3.cpp
+++ b/extlibs/libQGLViewer-2.6.3/QGLViewer/VRender/Vector3.cpp
@@ -45,6 +45,7 @@
 #include <iostream>
 #include "Vector3.h"
 #include "NVector3.h"
+#include <algorithm>
 #include <math.h>
 
 #ifdef WIN32
