diff --git a/SofaKernel/SofaFramework/CMakeLists.txt b/SofaKernel/SofaFramework/CMakeLists.txt
index b3b56c1ef1..bd353a40d7 100644
--- a/SofaKernel/SofaFramework/CMakeLists.txt
+++ b/SofaKernel/SofaFramework/CMakeLists.txt
@@ -73,7 +73,7 @@ if(NOT SOFA_NO_OPENGL)
         list(APPEND SOFAFRAMEWORK_DEPENDENCY_LINK ${GLEW_Target})
         list(APPEND SOFAFRAMEWORK_DEPENDENCY_INCLUDE_DIRECTORIES ${GLEW_INCLUDE_DIRS})
 
-        sofa_install_get_libraries(${GLEW_LIBRARIES})
+        #sofa_install_get_libraries(${GLEW_LIBRARIES})
     else()
         message("OpenGL advanced functions (e.g shaders, FBO) are disabled.")
     endif()
@@ -111,7 +111,7 @@ if(ZLIB_FOUND)
     sofa_create_target(ZLIB SofaFramework "${ZLIB_LIBRARIES}" "${ZLIB_INCLUDE_DIRS}")
     list(APPEND SOFAFRAMEWORK_DEPENDENCY_LINK ${ZLIB_Target})
 
-    sofa_install_get_libraries(${ZLIB_LIBRARIES})
+    #sofa_install_get_libraries(${ZLIB_LIBRARIES})
 endif()
 set(SOFA_HAVE_ZLIB ${ZLIB_FOUND})
 
@@ -122,7 +122,7 @@ if(PNG_FOUND)
         list(APPEND SOFAFRAMEWORK_DEPENDENCY_LINK ${PNG_Target})
 
 
-    sofa_install_get_libraries("${PNG_LIBRARY}/libpng")
+    #sofa_install_get_libraries("${PNG_LIBRARY}/libpng")
 endif()
 set(SOFA_HAVE_PNG ${PNG_FOUND})
 
@@ -153,20 +153,20 @@ if(SOFA_HAVE_BOOST_SYSTEM
     list(APPEND SOFAFRAMEWORK_DEPENDENCY_LINK ${BoostLocale_Target})
     list(APPEND SOFAFRAMEWORK_DEPENDENCY_LINK ${BoostFileSystem_Target})
 
-    sofa_install_get_libraries(${Boost_SYSTEM_LIBRARY})
-    sofa_install_get_libraries(${Boost_LOCALE_LIBRARY})
-    sofa_install_get_libraries(${Boost_FILESYSTEM_LIBRARY})
+    #sofa_install_get_libraries(${Boost_SYSTEM_LIBRARY})
+    #sofa_install_get_libraries(${Boost_LOCALE_LIBRARY})
+    #sofa_install_get_libraries(${Boost_FILESYSTEM_LIBRARY})
 
     if(SOFA_HAVE_BOOST_DATE_TIME)
         sofa_create_target(BoostDateTime SofaFramework "${Boost_DATE_TIME_LIBRARY}" "${Boost_INCLUDE_DIRS}")
         list(APPEND SOFAFRAMEWORK_DEPENDENCY_LINK ${BoostDateTime_Target})
-        sofa_install_get_libraries(${Boost_DATE_TIME_LIBRARY})
+        #sofa_install_get_libraries(${Boost_DATE_TIME_LIBRARY})
     endif()
 
     if(SOFA_HAVE_BOOST_THREAD)
         sofa_create_target(BoostThread SofaFramework "${Boost_THREAD_LIBRARY}" "${Boost_INCLUDE_DIRS}")
         list(APPEND SOFAFRAMEWORK_DEPENDENCY_LINK ${BoostThread_Target})
-        sofa_install_get_libraries(${Boost_THREAD_LIBRARY})
+        #sofa_install_get_libraries(${Boost_THREAD_LIBRARY})
     endif()
 
 else()
