diff --git a/boost/graph/detail/adjacency_list.hpp b/boost/graph/detail/adjacency_list.hpp
index 1145d88..96d08ca 100755
--- a/boost/graph/detail/adjacency_list.hpp
+++ b/boost/graph/detail/adjacency_list.hpp
@@ -296,7 +296,7 @@ namespace boost {
       inline stored_edge_property(Vertex target,
                                   const Property& p = Property())
         : stored_edge<Vertex>(target), m_property(new Property(p)) { }
-#if defined(BOOST_MSVC) || (defined(BOOST_GCC) && (BOOST_GCC / 100) < 406)
+#if defined(BOOST_MSVC) || defined(BOOST_GCC) || defined(BOOST_CLANG)
       stored_edge_property(self&& x) : Base(static_cast< Base const& >(x)) {
         m_property.swap(x.m_property);
       }
