diff --git a/qtmultimedia/src/plugins/gstreamer/camerabin/camerabinsession.cpp b/qtmultimedia/src/plugins/gstreamer/camerabin/camerabinsession.cpp
index f06d95ea3..4941c6ef6 100644
--- a/qtmultimedia/src/plugins/gstreamer/camerabin/camerabinsession.cpp
+++ b/qtmultimedia/src/plugins/gstreamer/camerabin/camerabinsession.cpp
@@ -729,21 +729,18 @@ void CameraBinSession::setState(QCamera::State newState)
     if (newState == m_pendingState)
         return;
 
-    emit pendingStateChanged(newState);
+    m_pendingState = newState;
+    emit pendingStateChanged(m_pendingState);
 
 #if CAMERABIN_DEBUG
     qDebug() << Q_FUNC_INFO << newState;
 #endif
 
     setStateHelper(newState);
-    m_pendingState = newState;
 }
 
 void CameraBinSession::setStateHelper(QCamera::State state)
 {
-    if (state == m_pendingState)
-        return;
-
     switch (state) {
     case QCamera::UnloadedState:
         unload();
