diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee9f1d1..51f3ce6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,24 +1,5 @@
 cmake_minimum_required(VERSION 2.8)
 
-set(CUDA_ARCH "-arch=sm_50" CACHE STRING "Value of the NVCC -arch option.")
-
-option(ENABLE_ZED_DEMO "Build a Demo using ZED Camera" OFF)
-
-if(WIN32)
-  set(ZED_SDK_LIB "C:\\Program Files (x86)\\ZED SDK\\lib\\sl_zed64.lib" CACHE STRING "ZED SDK library(sl_zed**.llb) path.")
-  set(ZED_SDK_INCLUDE_DIR "C:\\Program Files (x86)\\ZED SDK\\include" CACHE STRING "ZED SDK include path.")
-else()
-  set(ZED_SDK_LIB "/usr/local/zed/lib/libsl_zed.so" CACHE STRING "ZED SDK library(sl_zed**.llb) path.")
-  set(ZED_SDK_INCLUDE_DIR "/usr/local/zed/include" CACHE STRING "ZED SDK include path.")
-endif()
-
 project(libSGM)
 
 add_subdirectory(src)
-
-add_subdirectory(sample/image)
-add_subdirectory(sample/movie)
-
-if(ENABLE_ZED_DEMO)
-	add_subdirectory(sample/zed)
-endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index af93aff..e86eb03 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,8 +1,6 @@
-find_package(CUDA REQUIRED)
-
 include_directories(../include)
 
-SET(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} ${CUDA_ARCH}")
+find_package(CUDA REQUIRED)
 
 SET(STEREOSRCS 
 	census.cu 
@@ -14,4 +12,16 @@ SET(STEREOSRCS
 	cuda_utils.cu
 )
 
-CUDA_ADD_LIBRARY(sgm stereo_sgm.cpp ${STEREOSRCS} STATIC)
+CUDA_ADD_LIBRARY(sgm stereo_sgm.cpp ${STEREOSRCS} SHARED)
+
+install(
+    TARGETS sgm
+    LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
+)
+
+install(
+    DIRECTORY ${CMAKE_SOURCE_DIR}/include
+    DESTINATION ${CMAKE_INSTALL_PREFIX}
+    FILES_MATCHING PATTERN "*.h" PATTERN "*.hpp"
+)
