diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7eee54a..d8f1dd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ include(ExternalProject)
 if( CMAKE_HOST_WIN32 )
 
   string( LENGTH "${CMAKE_CURRENT_SOURCE_DIR}" n )
-  if( n GREATER 50 )
+  if( n GREATER 500 )
     message(
       FATAL_ERROR
       "ITK source code directory path length is too long (${n} > 50)."
@@ -32,7 +32,7 @@ if( CMAKE_HOST_WIN32 )
   endif()
 
   string( LENGTH "${CMAKE_CURRENT_BINARY_DIR}" n )
-  if( n GREATER 50 )
+  if( n GREATER 500 )
     message(
       FATAL_ERROR
       "ITK build directory path length is too long (${n} > 50)."
