diff -ru a/Code/IO/itkDefaultConvertPixelTraits.h b/Code/IO/itkDefaultConvertPixelTraits.h
--- a/Code/IO/itkDefaultConvertPixelTraits.h	2011-10-25 18:37:23.000000000 +0200
+++ b/Code/IO/itkDefaultConvertPixelTraits.h	2014-02-28 19:59:11.000000000 +0100
@@ -291,11 +291,11 @@
 ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(float);
 ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(double);
 ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(signed int);
-ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(unsigned int);
+// ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(unsigned int);
 ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(signed char);
 ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(unsigned char);
 ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(signed long);
-ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(unsigned long);
+// ITK_DEFAULTCONVERTTRAITS_COMPLEX_TYPE(unsigned long);
 
 //
 //  End of Traits for the classes deriving from std::complex.
