diff -u -r a/Code/IO/CMakeLists.txt b/Code/IO/CMakeLists.txt
--- a/Code/IO/CMakeLists.txt	2011-10-25 18:37:23.000000000 +0200
+++ b/Code/IO/CMakeLists.txt	2013-11-15 16:56:56.000000000 +0100
@@ -97,7 +97,7 @@
 ENDIF( ITK_USE_TRANSFORM_IO_FACTORIES )
 
 IF(CMAKE_COMPILER_IS_GNUCXX)
-  SET_SOURCE_FILES_PROPERTIES(itkTestDriver.cxx PROPERTIES COMPILE_FLAGS -w)
+  # SET_SOURCE_FILES_PROPERTIES(itkTestDriver.cxx PROPERTIES COMPILE_FLAGS -w)
   SET_SOURCE_FILES_PROPERTIES(itkPolygonGroupSpatialObjectXMLFile.cxx PROPERTIES COMPILE_FLAGS -w)
   SET_SOURCE_FILES_PROPERTIES(itkJPEGImageIO.cxx PROPERTIES COMPILE_FLAGS -w)
   SET_SOURCE_FILES_PROPERTIES(itkPNGImageIO.cxx PROPERTIES COMPILE_FLAGS -w)
@@ -113,14 +113,14 @@
 # itkTestDriver is built even when BUILD_TESTING is FALSE to:
 # + allow to run WrapITK tests without the other of the ITK tests
 # + be used by external code like contributions to the insight journal
-ADD_EXECUTABLE(itkTestDriver itkTestDriver.cxx) 
-TARGET_LINK_LIBRARIES(itkTestDriver ITKIO)
-IF(NOT ITK_INSTALL_NO_DEVELOPMENT)
-  INSTALL(TARGETS itkTestDriver
-    RUNTIME DESTINATION ${ITK_INSTALL_BIN_DIR_CM24} COMPONENT Development)
-ENDIF(NOT ITK_INSTALL_NO_DEVELOPMENT)
-SET(ITK_TEST_DRIVER "${EXECUTABLE_OUTPUT_PATH}/itkTestDriver" 
-    CACHE INTERNAL "itkTestDriver path to be used by subprojects")
+# ADD_EXECUTABLE(itkTestDriver itkTestDriver.cxx) 
+# TARGET_LINK_LIBRARIES(itkTestDriver ITKIO)
+# IF(NOT ITK_INSTALL_NO_DEVELOPMENT)
+#   INSTALL(TARGETS itkTestDriver
+#     RUNTIME DESTINATION ${ITK_INSTALL_BIN_DIR_CM24} COMPONENT Development)
+# ENDIF(NOT ITK_INSTALL_NO_DEVELOPMENT)
+# SET(ITK_TEST_DRIVER "${EXECUTABLE_OUTPUT_PATH}/itkTestDriver" 
+#     CACHE INTERNAL "itkTestDriver path to be used by subprojects")
 
 IF(ITK_LIBRARY_PROPERTIES)
   SET_TARGET_PROPERTIES(ITKIO PROPERTIES ${ITK_LIBRARY_PROPERTIES})
Only in InsightToolkit-3.20.1/Code/IO: CMakeLists.txt.rej
