diff --git a/PCLConfig.cmake.in b/PCLConfig.cmake.in
index 6350a87b7..db8f46500 100644
--- a/PCLConfig.cmake.in
+++ b/PCLConfig.cmake.in
@@ -274,6 +274,10 @@ macro(find_external_library _component _lib _is_optional)
     find_boost()
   elseif("${_lib}" STREQUAL "eigen")
     find_eigen()
+    if(NOT EIGEN_FOUND AND EIGEN3_FOUND)
+      set(EIGEN_FOUND ${EIGEN3_FOUND})
+      set(EIGEN_INCLUDE_DIRS ${EIGEN3_INCLUDE_DIR})
+    endif()
   elseif("${_lib}" STREQUAL "flann")
     find_flann()
   elseif("${_lib}" STREQUAL "qhull")
@@ -569,7 +573,10 @@ endif(NOT "${PCL_DEFINITIONS}" STREQUAL "")
 pcl_remove_duplicate_libraries(PCL_LIBRARIES PCL_DEDUP_LIBRARIES)
 set(PCL_LIBRARIES ${PCL_DEDUP_LIBRARIES})
 # Add 3rd party libraries, as user code might include our .HPP implementations
-list(APPEND PCL_LIBRARIES ${BOOST_LIBRARIES} ${QHULL_LIBRARIES} ${OPENNI_LIBRARIES} ${OPENNI2_LIBRARIES} ${ENSENSO_LIBRARIES} ${davidSDK_LIBRARIES} ${DSSDK_LIBRARIES} ${RSSDK_LIBRARIES} ${FLANN_LIBRARIES} ${VTK_LIBRARIES})
+list(APPEND PCL_LIBRARIES ${BOOST_LIBRARIES} ${OPENNI_LIBRARIES} ${OPENNI2_LIBRARIES} ${ENSENSO_LIBRARIES} ${davidSDK_LIBRARIES} ${DSSDK_LIBRARIES} ${RSSDK_LIBRARIES} ${FLANN_LIBRARIES} ${VTK_LIBRARIES})
+if(QHULL_LIBRARIES)
+    list(APPEND PCL_LIBRARIES ${QHULL_LIBRARIES})
+endif()
 
 find_package_handle_standard_args(PCL DEFAULT_MSG PCL_LIBRARIES PCL_INCLUDE_DIRS)
 mark_as_advanced(PCL_LIBRARIES PCL_INCLUDE_DIRS PCL_LIBRARY_DIRS)
