diff -uNr a/cmake/pcl_targets.cmake b/cmake/pcl_targets.cmake
--- a/cmake/pcl_targets.cmake	2016-06-14 17:51:26.000000000 +0200
+++ b/cmake/pcl_targets.cmake	2016-07-21 10:25:25.908049000 +0200
@@ -196,7 +196,7 @@
       target_link_libraries(${_name} gomp)
     endif()
 	
-	if(MSVC90 OR MSVC10)
+	if(MSVC90 OR MSVC10 OR MSVC12)
 	  target_link_libraries(${_name} delayimp.lib)  # because delay load is enabled for openmp.dll
 	endif()
 
diff -uNr a/visualization/src/pcl_visualizer.cpp b/visualization/src/pcl_visualizer.cpp
--- a/visualization/src/pcl_visualizer.cpp	2016-06-14 17:51:26.000000000 +0200
+++ b/visualization/src/pcl_visualizer.cpp	2016-07-21 10:39:13.825865000 +0200
@@ -1492,7 +1492,7 @@
       actor->GetMapper ()->ScalarVisibilityOn ();
       actor->GetMapper ()->SetScalarRange (range[0], range[1]);
       vtkSmartPointer<vtkLookupTable> table;
-      if (!pcl::visualization::getColormapLUT (static_cast<LookUpTableRepresentationProperties>(value), table))
+      if (!pcl::visualization::getColormapLUT (static_cast<LookUpTableRepresentationProperties>(static_cast<int>(value)), table))
         break;
       table->SetRange (range[0], range[1]);
       actor->GetMapper ()->SetLookupTable (table);
@@ -1738,7 +1738,7 @@
       actor->GetMapper ()->ScalarVisibilityOn ();
       actor->GetMapper ()->SetScalarRange (range[0], range[1]);
       vtkSmartPointer<vtkLookupTable> table = vtkSmartPointer<vtkLookupTable>::New ();
-      getColormapLUT (static_cast<LookUpTableRepresentationProperties>(value), table);
+      getColormapLUT (static_cast<LookUpTableRepresentationProperties>(static_cast<int>(value)), table);
       table->SetRange (range[0], range[1]);
       actor->GetMapper ()->SetLookupTable (table);
       style_->updateLookUpTableDisplay (false);
