diff -uNr a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt    2017-05-30 08:57:09.012280746 +0200
+++ b/CMakeLists.txt    2017-05-30 08:56:37.368664755 +0200
@@ -210,7 +210,7 @@
 endif()
 
 # default to not using the system GLEW as ours has fixes in it right now
-set(VTK_USE_SYSTEM_GLEW OFF CACHE BOOL "do not use a system glew" FORCE)
+set(VTK_USE_SYSTEM_GLEW OFF CACHE BOOL "do not use a system glew" )
 
 set(VTK_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
 
