diff -uNr a/CMake/vtkOpenGL.cmake b/CMake/vtkOpenGL.cmake
--- a/CMake/vtkOpenGL.cmake	2017-07-27 09:08:48.000000000 +0200
+++ b/CMake/vtkOpenGL.cmake	2017-07-27 09:14:06.000000000 +0200
@@ -102,6 +102,6 @@
     vtk_module_link_libraries(${target} LINK_PRIVATE ${EGL_LIBRARIES})
   endif()
   if(VTK_CAN_DO_ONSCREEN)
-    vtk_module_link_libraries(${target} LINK_PRIVATE ${OPENGL_LIBRARIES})
+    vtk_module_link_libraries(${target} LINK_PRIVATE ${OPENGL_LIBRARIES} ${GLEW_LIBRARY})
   endif()
 endfunction()