--- a/Rendering/OpenGL/vtkPixelBufferObject.cxx	2016-11-14 16:05:14.000000000 +0100
+++ b/Rendering/OpenGL/vtkPixelBufferObject.cxx	2017-03-16 14:28:35.860670400 +0100
@@ -321,52 +321,77 @@
 
       int numComp;
       int *permutation=0;
+
+      // If no component reordering is necessary, we can make some optimization
       if(components==0)
       {
-        numComp=numComponents;
-        permutation=new int[numComponents];
-        int i=0;
-        while(i<numComp)
+        // Shortest path, everything is contiguous in memory
+        // (tested with a 1980x1080x1, 4 bytes per component, RGBA image: 80% faster in release, 97% faster in debug)
+        if(continuousIncrements[0] == 0 && continuousIncrements[1] == 0 && continuousIncrements[2] == 0)
+        {
+            memcpy(fIoMem, inData, numComponents * dims[0] * dims[1] * dims[2]);
+        }
+        else
         {
-          permutation[i]=i;
-          ++i;
+          // This is not contiguous in memory but we can skip the reordering
+          // (tested with a 1980x1080x1, 4 bytes per component, image: 0% faster in release but 65% faster in debug)
+          vtkIdType tupleSize =
+            static_cast<vtkIdType>(numComponents + continuousIncrements[0]);
+          for (unsigned int zz=0; zz < dims[2]; zz++)
+            {
+            for (unsigned int yy = 0; yy < dims[1]; yy++)
+              {
+              for (unsigned int xx=0; xx < dims[0]; xx++)
+                {
+                memcpy(fIoMem, inData, tupleSize);
+
+                fIoMem += tupleSize;
+                inData += tupleSize+continuousIncrements[0];
+                }
+              // Reached end of row, go to start of next row.
+              inData += continuousIncrements[1] * tupleSize;
+              }
+            // Reached end of 2D plane.
+            inData += continuousIncrements[2] * tupleSize;
+            }
         }
       }
       else
       {
+        // Generic path
         numComp=components;
         permutation=componentList;
-      }
 
-      vtkIdType tupleSize =
-        static_cast<vtkIdType>(numComponents + continuousIncrements[0]);
-      for (unsigned int zz=0; zz < dims[2]; zz++)
-      {
-        for (unsigned int yy = 0; yy < dims[1]; yy++)
-        {
-          for (unsigned int xx=0; xx < dims[0]; xx++)
+          vtkIdType tupleSize =
+            static_cast<vtkIdType>(numComponents + continuousIncrements[0]);
+          for (unsigned int zz=0; zz < dims[2]; zz++)
           {
-            for (int compNo=0; compNo < numComp; compNo++)
+            for (unsigned int yy = 0; yy < dims[1]; yy++)
             {
-              *fIoMem = inData[permutation[compNo]];
-//              cout<<"upload[zz="<<zz<<"][yy="<<yy<<"][xx="<<xx<<"][compNo="<<
-//              compNo<<"] from inData to pbo="<<(double)(*fIoMem)<<endl;
+              for (unsigned int xx=0; xx < dims[0]; xx++)
+              {
+                for (int compNo=0; compNo < numComp; compNo++)
+                {
+                  *fIoMem = inData[permutation[compNo]];
+    //              cout<<"upload[zz="<<zz<<"][yy="<<yy<<"][xx="<<xx<<"][compNo="<<
+    //              compNo<<"] from inData to pbo="<<(double)(*fIoMem)<<endl;
 
-              fIoMem++;
+                  fIoMem++;
+                }
+                inData += tupleSize+continuousIncrements[0];
+              }
+              // Reached end of row, go to start of next row.
+              inData += continuousIncrements[1] * tupleSize;
             }
-            inData += tupleSize+continuousIncrements[0];
+            // Reached end of 2D plane.
+            inData += continuousIncrements[2] * tupleSize;
           }
-          // Reached end of row, go to start of next row.
-          inData += continuousIncrements[1] * tupleSize;
-        }
-        // Reached end of 2D plane.
-        inData += continuousIncrements[2] * tupleSize;
-      }
 
       if(components==0)
       {
         delete[] permutation;
       }
+    }
   }
 };
 
