--- a/Common/Core/vtkAndroidOutputWindow.cxx
+++ b/Common/Core/vtkAndroidOutputWindow.cxx
@@ -45,7 +45,7 @@ void vtkAndroidOutputWindow::DisplayErrorText(const char* someText)
   std::string line;
   while (std::getline(stream, line))
     {
-    __android_log_print(ANDROID_LOG_ERROR, "VTK", line.c_str());
+    __android_log_print(ANDROID_LOG_ERROR, "VTK", "%s", line.c_str());
     }
   this->InvokeEvent(vtkCommand::ErrorEvent, (void*)someText);
 }
@@ -62,7 +62,7 @@ void vtkAndroidOutputWindow::DisplayWarningText(const char* someText)
   std::string line;
   while (std::getline(stream, line))
     {
-    __android_log_print(ANDROID_LOG_WARN, "VTK", line.c_str());
+    __android_log_print(ANDROID_LOG_WARN, "VTK", "%s", line.c_str());
     }
   this->InvokeEvent(vtkCommand::WarningEvent,(void*) someText);
 }
@@ -79,7 +79,7 @@ void vtkAndroidOutputWindow::DisplayGenericWarningText(const char* someText)
   std::string line;
   while (std::getline(stream, line))
     {
-    __android_log_print(ANDROID_LOG_WARN, "VTK", line.c_str());
+    __android_log_print(ANDROID_LOG_WARN, "VTK", "%s", line.c_str());
     }
 }

@@ -95,7 +95,7 @@ void vtkAndroidOutputWindow::DisplayDebugText(const char* someText)
   std::string line;
   while (std::getline(stream, line))
     {
-    __android_log_print(ANDROID_LOG_DEBUG, "VTK", line.c_str());
+    __android_log_print(ANDROID_LOG_DEBUG, "VTK", "%s", line.c_str());
     }
 }

@@ -111,7 +111,7 @@ void vtkAndroidOutputWindow::DisplayText(const char* someText)
   std::string line;
   while (std::getline(stream, line))
     {
-    __android_log_print(ANDROID_LOG_INFO, "VTK", line.c_str());
+    __android_log_print(ANDROID_LOG_INFO, "VTK", "%s", line.c_str());
     }
 }
