diff -uNr a/Source/Drivers/OniFile/Makefile /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/OniFile/Makefile
--- a/Source/Drivers/OniFile/Makefile	2017-05-03 11:28:13.951590659 +0200
+++ /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/OniFile/Makefile	2017-05-03 11:45:22.565903719 +0200
@@ -24,8 +24,9 @@
 
 LIB_NAME = OniFile
 
-LIB_DIRS = ../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG)
+LIB_DIRS = ../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG) \
+		   $(INSTALL_LIB_DIR)
 USED_LIBS = XnLib pthread dl
 ifneq ("$(OSTYPE)","Darwin")
         USED_LIBS += rt  
 endif
diff -uNr a/Source/Drivers/orbbec/Makefile /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/orbbec/Makefile
--- a/Source/Drivers/orbbec/Makefile	2017-05-03 11:28:13.963590500 +0200
+++ /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/orbbec/Makefile	2017-05-03 11:44:43.050429520 +0200
@@ -32,7 +32,8 @@
 
 LIB_NAME = orbbec
 
-LIB_DIRS += ../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG)
+LIB_DIRS += ../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG) \
+		   $(INSTALL_LIB_DIR)
 LIB_DIRS += $(BIN_DIR)/$(PLATFORM)-$(CFG)
 USED_LIBS = XnLib dl pthread DepthUtils
 ifneq ("$(OSTYPE)","Darwin")
diff -uNr a/Source/Drivers/PS1080/Makefile /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/PS1080/Makefile
--- a/Source/Drivers/PS1080/Makefile	2017-05-03 11:28:13.955590606 +0200
+++ /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/PS1080/Makefile	2017-05-03 11:44:34.370545016 +0200
@@ -32,7 +32,8 @@
 
 LIB_NAME = PS1080
 
-LIB_DIRS += ../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG)
+LIB_DIRS += ../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG) \
+		   $(INSTALL_LIB_DIR)
 LIB_DIRS += $(BIN_DIR)/$(PLATFORM)-$(CFG)
 USED_LIBS = XnLib dl pthread DepthUtils
 ifneq ("$(OSTYPE)","Darwin")
diff -uNr a/Source/Drivers/PS1080/PS1080Console/Makefile /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/PS1080/PS1080Console/Makefile
--- a/Source/Drivers/PS1080/PS1080Console/Makefile	2017-05-03 11:28:13.959590553 +0200
+++ /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/PS1080/PS1080Console/Makefile	2017-05-03 11:45:58.217429333 +0200
@@ -10,7 +10,8 @@
 SRC_FILES = \
 	*.cpp \
 
-LIB_DIRS = ../../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG)
+LIB_DIRS = ../../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG) \
+		   $(INSTALL_LIB_DIR)
 USED_LIBS = XnLib OpenNI2 dl pthread
 
 ifeq ("$(OSTYPE)","Darwin")
diff -uNr a/Source/Drivers/PSLink/Makefile /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/PSLink/Makefile
--- a/Source/Drivers/PSLink/Makefile	2017-05-03 11:28:13.951590659 +0200
+++ /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/PSLink/Makefile	2017-05-03 11:45:02.530170319 +0200
@@ -14,8 +14,9 @@
 	DriverImpl/*.cpp \
 	LinkProtoLib/*.cpp
 
-LIB_DIRS  = ../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG)
+LIB_DIRS  = ../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG) \
+		   $(INSTALL_LIB_DIR)
 LIB_DIRS += $(BIN_DIR)/$(PLATFORM)-$(CFG)
 USED_LIBS = XnLib dl pthread
 
 ifeq ("$(OSTYPE)","Darwin")
diff -uNr a/Source/Drivers/PSLink/PSLinkConsole/Makefile /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/PSLink/PSLinkConsole/Makefile
--- a/Source/Drivers/PSLink/PSLinkConsole/Makefile	2017-05-03 11:28:13.955590606 +0200
+++ /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Drivers/PSLink/PSLinkConsole/Makefile	2017-05-03 11:46:15.969193125 +0200
@@ -14,7 +14,8 @@
 	../LinkProtoLib/*.cpp \
 	../*.cpp \
 
-LIB_DIRS = ../../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG)
+LIB_DIRS = ../../../../ThirdParty/PSCommon/XnLib/Bin/$(PLATFORM)-$(CFG) \
+		   $(INSTALL_LIB_DIR)
 USED_LIBS = OpenNI2 XnLib dl pthread
 
 ifeq ("$(OSTYPE)","Darwin")
diff -uNr a/Source/Tools/NiViewer/Makefile /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Tools/NiViewer/Makefile
--- a/Source/Tools/NiViewer/Makefile	2017-05-03 11:28:13.971590393 +0200
+++ /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/Source/Tools/NiViewer/Makefile	2017-05-03 11:37:17.048364109 +0200
@@ -7,6 +7,8 @@
 	../../../ThirdParty/PSCommon/XnLib/Include \
 	../../../ThirdParty/GL/
 
+LIB_DIRS += $(INSTALL_LIB_DIR)
+
 SRC_FILES = \
 	Device.cpp  \
 	Draw.cpp  \
diff -uNr a/ThirdParty/PSCommon/XnLib/Source/Makefile /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/ThirdParty/PSCommon/XnLib/Source/Makefile
--- a/ThirdParty/PSCommon/XnLib/Source/Makefile	2017-05-03 11:28:13.919591085 +0200
+++ /home/fbridault/dev/f4s/deps/build/debug/openni/openni-prefix/src/openni/ThirdParty/PSCommon/XnLib/Source/Makefile	2017-05-03 11:37:17.048364109 +0200
@@ -2,7 +2,7 @@
 
 BIN_DIR = ../Bin
 
-INC_DIRS = \
+INC_DIRS = $(INSTALL_INCLUDE_DIR) \
 	../Include \
 	../ThirdParty/LibJPEG

